code/modules/power/cell.dm

```diff
@@ -118,9 +118,7 @@
 		rigged = 1 //broken batterys are dangerous
 
 /obj/item/weapon/cell/emp_act(severity)
-	charge -= 1000 / severity
-	if (charge < 0)
-		charge = 0
+	charge = max(charge - maxcharge/severity, 0)
```

Comment:
Trying it out it does seem a tad unbalanced, choosing a way that scales appropriately from 1,000 charge all the way up to 30,000 charge is somewhat challenging.